platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookValidator.java

```diff
@@ -151,4 +143,35 @@ && isNonEmpty(candidateAddressBook)
                 && validNextId(previousAddressBook, candidateAddressBook)
                 && noAddressReinsertion(previousAddressBook, candidateAddressBook);
     }
+
+    /**
+     * Checks that the addresses between the two address books are identical except for stake value.
+     *
+     * @param addressBook1 An address book to compare for equality.
+     * @param addressBook2 An address book to compare for equality.
+     * @return true of the two address books contain the same addresses except for stake values, false otherwise.
+     */
+    public static boolean sameExceptForStake(AddressBook addressBook1, AddressBook addressBook2) {
+        int addressBookSize = addressBook1.getSize();
+        return addressBookSize == addressBook2.getSize()
+                && IntStream.range(0, addressBookSize)
+                        .mapToObj(i -> {
+                            final Address address1 = addressBook1.getAddress(i);
+                            final Address address2 = addressBook2.getAddress(i);
+                            if (address1 == null || address2 == null) {
+                                return address1 == address2;
+                            } else {
```

Comment:
Prior job conditioned me to always have an else clause to disambiguates intent in debugging/review scenarios.  "was the content after the if supposed to be in an else?"  "should there be an else clause?"   "Was an else clause left off by mistake?"   --  I'll leave it in since this does not match the pattern for [guard blocks](https://softwareengineering.stackexchange.com/questions/157407/best-practice-on-if-return).